src/rust/engine/process_execution/src/remote.rs

```diff
@@ -633,6 +636,27 @@ fn format_error(error: &bazel_protos::status::Status) -> String {
   format!("{}: {}", error_code, error.get_message())
 }
 
+///
+/// If the given operation represents a cancelled request, recover it into
+/// ExecutionError::NotFinished.
+///
+fn rpcerror_recover_cancelled(
+  operation_name: String,
+  err: grpcio::Error,
+) -> Result<bazel_protos::operations::Operation, grpcio::Error> {
+  // If the error represented cancellation, return an Operation for the given Operation name.
+  match &err {
+    &grpcio::Error::RpcFailure(ref rs) if rs.status == grpcio::RpcStatusCode::Cancelled => {
+      let mut next_operation = bazel_protos::operations::Operation::new();
+      next_operation.set_name(operation_name);
+      return Ok(next_operation);
+    }
+    _ => {}
+  }
+  // Did not represent cancellation.
+  Err(err)
```

Comment:
This could just be the RHS of the `_` match clause?

Reply:
That would complicate things, because I'd have to take the match by value, which means I could not have a guard clause on that branch of the match.